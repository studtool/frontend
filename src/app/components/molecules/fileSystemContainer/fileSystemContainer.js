import React, {Component} from 'react';
import {FileSystemElement} from '../../atoms/fileSystemElement/fileSystemElement.js';

export class FileSystemContainer extends Component {
    constructor(props) {
        super(props);

        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleSubmit(e){
        if (e.keyCode == 13){
            e.preventDefault();
            console.log(e.nativeEvent.srcElement.innerText);
        }
    }

    render() {
        const {
            qa = false,
            data = {},
            children = null,
            handleSubmit = this.handleSubmit,
        } = this.props;

        return (
            <div data-qa={qa ? 'file-system-container' : void 0}>
                <FileSystemElement handleSubmit={handleSubmit}>
                    {children}
                </FileSystemElement>
            </div>
        );
    };
}
