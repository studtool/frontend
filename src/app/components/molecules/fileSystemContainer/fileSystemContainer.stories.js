import * as React from 'react';
import {storiesOf} from '@storybook/react';
import {FileSystemContainer} from 'App/components/molecules/fileSystemContainer/fileSystemContainer';

storiesOf('File system container ', module)
    .add('default', () => (
        <FileSystemContainer>
            Название файла
        </FileSystemContainer>
    ))
    .add('with qa', () => (
        <FileSystemContainer qa={true}/>
    ))
