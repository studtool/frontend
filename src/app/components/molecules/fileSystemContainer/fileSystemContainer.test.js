import React from 'react';
import {FileSystemContainer} from './fileSystemContainer';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

test('file System Container test', () => {
    const stubSubmit = () => {};
    const stubChange = () => {};
    const data = {
        email__errorMessage: '',
        password__errorMessage: '',
        passwordRepeat__errorMessage: '',
    };

    const tree = renderer.create(
        <FileSystemContainer
            handleSubmit={stubSubmit}
            handleChange={stubChange}
            data={data}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
