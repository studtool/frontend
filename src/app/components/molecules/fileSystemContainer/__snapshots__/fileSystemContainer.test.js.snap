// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`file System Container test 1`] = `
<div>
  <div
    contentEditable={true}
    onKeyDown={[Function]}
  />
</div>
`;
